File: data/lms_verify/ex2.check.c
#include <limits.h>
/*@
requires (x0>0);
assigns \nothing;
ensures ((0<=\result) &&
(\result<x0));
*/
int pick_index(int  x0) {
  return 0;
}
/*@
requires ((x7>0) &&
\valid(x6+(0..x7-1)));
assigns \nothing;
*/
int pick_element(int  * x6, int  x7) {
  int x9 = pick_index(x7);
  int x10 = x6[x9];
  return x10;
}
/*@
requires \valid(x15);
assigns \nothing;
ensures (\result==x15[0]);
*/
int pick_first(int  * x15) {
  int x17 = x15[0];
  return x17;
}

File: transpiled_rust/lms_verify/ex2_check.rs
fn pick_index(x0: i32) -> i32 {
    0
}

fn pick_element(x6: &[i32], x7: i32) -> i32 {
    let x9: i32 = pick_index(x7);
    let x10: i32 = x6[x9 as usize];
    x10
}

fn pick_first(x15: &[i32]) -> i32 {
    let x17: i32 = x15[0];
    x17
}